dart/lib/src/protocol/event_processor.dart

```diff
@@ -0,0 +1,13 @@
+import 'dart:async';
+
+import '../protocol.dart';
+
+/// [EventProcessor]s are callbacks that run for every event. They can either
+/// return a new event which in most cases means just adding data *or* return
+/// null in case the event will be dropped and not sent.
+abstract class EventProcessor {
```

Comment:
event processor is part of the unified API but not the protocol, protocol is the data classes, so I'd move it